packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2607,6 +2609,16 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       if (this.state.isAmbientContext && this.match(tt.eq)) {
         this.raise(this.state.start, TSErrors.DeclareClassFieldHasInitializer);
       }
+      if (node.abstract && this.match(tt.eq)) {
+        const { key } = node;
+        this.raise(
+          this.state.start,
+          TSErrors.AbstractPropertyHasInitializer,
+          key.type === "Identifier"
```

Comment:
```suggestion
          key.computed === false
```

An identifier key can be computed, too.

Reply:
Do we have to change it for abstract methods too, then?

Reply:
We could also throw for abstract methods, though the case is rare since the normal method does not allow initializers, too.

Reply:
If you are talking about https://github.com/babel/babel/blob/4ee78eb3f194d6015af145060d9b61794d1ea477/packages/babel-parser/src/plugins/typescript/index.js#L3185-L3187, yes we should change it :+1: